File: source/com/barrybecker4/puzzle/sudoku/model/board/CellArrays.java
/** Copyright by Barry G. Becker, 2000-2011. Licensed under MIT License: http://www.opensource.org/licenses/MIT  */
package com.barrybecker4.puzzle.sudoku.model.board;


import java.util.Arrays;

/**
 *  All the arrays of cells for all the rows or columns in the puzzle.
 *
 *  @author Barry Becker
 */
public class CellArrays {

    /** candidate sets for a row or col.   */
    private CellArray[] cellArrays_;

    private int size;

    /**
     * Constructor
     */
    private CellArrays(int size) {

        this.size = size;
        cellArrays_ = new CellArray[size];
    }

    public static CellArrays createRowCellArrays(Board board) {

        CellArrays cellArrays = new CellArrays(board.getEdgeLength());
        for (int i=0; i<board.getEdgeLength(); i++) {
            cellArrays.cellArrays_[i] = CellArray.createRowCellArray(i, board);
        }
        return cellArrays;
    }

    public static CellArrays createColCellArrays(Board board) {
        CellArrays cellArrays = new CellArrays(board.getEdgeLength());
        for (int i=0; i<board.getEdgeLength(); i++) {
            cellArrays.cellArrays_[i] = CellArray.createColCellArray(i, board);
        }
        return cellArrays;
    }

    public CellArray get(int i) {
        return cellArrays_[i];
    }


    public void updateAll(ValuesList values) {

        for (int entry = 0; entry < size; entry++) {
            cellArrays_[entry].updateCandidates(values);
        }
    }

    public String toString() {
       return Arrays.toString(cellArrays_);
    }
}

File: scala-test/com/barrybecker4/puzzle/sudoku/model/board/CellArrays.scala
// Copyright by Barry G. Becker, 2017. Licensed under MIT License: http://www.opensource.org/licenses/MIT
package com.barrybecker4.puzzle.sudoku.model.board

/**
  * All the arrays of cells for all the rows or columns in the puzzle.
  *
  * @author Barry Becker
  */
object CellArrays {
  def createRowCellArrays(board: Board): CellArrays = {
    val cellArrays = new CellArrays(board.getEdgeLength)
    for (i <- 0 until board.getEdgeLength) {
        cellArrays.cellArrays(i) = CellArray.createRowCellArray(i, board)
    }
    cellArrays
  }

  def createColCellArrays(board: Board): CellArrays = {
    val cellArrays = new CellArrays(board.getEdgeLength)
    for (i <- 0 until board.getEdgeLength) {
        cellArrays.cellArrays(i) = CellArray.createColCellArray(i, board)
    }
    cellArrays
  }
}

class CellArrays private(var size: Int) {

  /** candidate sets for a row or col.   */
  private var cellArrays: Array[CellArray] = new Array[CellArray](n)

  def get(i: Int): CellArray = cellArrays(i)

  def updateAll(values: ValuesList) {
    for (entry <- 0 until size) {
        cellArrays(entry).updateCandidates(values)
    }
  }

  override def toString: String = cellArrays.mkString(", ")
}
